### PR TITLE
Filter empty newsletter months from publication

### PR DESCRIPTION
**Pull request by ewels**
## Summary
This PR prevents publishing empty newsletter pages by filtering out months that would render no content. Previously, the newsletter could generate pages with only headers and footers when content dates didn't align with the month boundaries used for newsletter generation.

## Key Changes
- **Added `newsletterMonthHasContent()` function**: A new utility that checks whether a newsletter for a given month would have any renderable content by examining all content sections (blog posts, advisories, pipelines, events, proposals) with their respective lookback/lookahead windows.

- **Added `getPublishedNewsletterMonths()` function**: A new public API that returns the canonical list of newsletter months with actual content, ensuring the listing page and RSS feed stay synchronized with generated pages.

- **Updated `getNewsletterStaticPathsData()`**: Now filters candidate months through `newsletterMonthHasContent()` to exclude empty pages before they're generated.

- **Updated newsletter listing page and RSS feed**: Both now use the new `getPublishedNewsletterMonths()` function instead of directly calling `getNewsletterMonths()`, ensuring they only reference months that actually exist.

- **Added comprehensive test coverage**: New test case validates the content gating logic, including edge cases like lone advisories creating orphaned pages and upcoming events spanning month boundaries.

## Implementation Details
The fix accounts for the newsletter's temporal offset: a newsletter dated the 1st of a month looks back at the previous calendar month for most content (blog posts, advisories, releases, pipelines) but forward at the current and next month for upcoming events. This means a month seeded by content can end up empty if that content only appears in the lookback/lookahead windows of adjacent months. The solution mirrors the section gating logic in `NewsletterLayout` to prevent publishing pages with no actual content.

https://claude.ai/code/session_016APe7N3CUvzw8qa1AtLkFv

@netlify /newsletter

### CODE REVIEW

File: sites/main-site/utils/newsletter.ts [hunk 1]
```diff
@@ -149,6 +149,44 @@ export function getNewsletterMonths(
...
+    blogPosts: { id: string; data: { pubDate: Date } }[],
+    events: { id: string; data: { start: Date } }[],
+    pipelines: PipelineWorkflow[],
+    advisories: { id: string; data: { publishedDate: Date } }[],
+    proposals: RawProposal[],
+    year: number,
+    month: number,
+): boolean {
+    const contentMonth = month === 1 ? 12 : month - 1;
+    const contentYear = month === 1 ? year - 1 : year;
```
Comment (mashehu) [inline sites/main-site/utils/newsletter.ts]: dates in javascript....

File: sites/main-site/utils/newsletter.ts [hunk 2]
```diff
@@ -149,6 +149,44 @@ export function getNewsletterMonths(
...
+        getBlogPostsForPreviousMonths(blogPosts, contentYear, contentMonth, 2).length > 0 ||
+        getAdvisoriesForMonth(advisories, contentYear, contentMonth).length > 0 ||
+        getAdvisoriesForPreviousMonths(advisories, contentYear, contentMonth, 2).length > 0 ||
+        getPipelineReleasesForMonth(pipelines, contentYear, contentMonth).length > 0 ||
+        getNewPipelines(pipelines, contentYear, contentMonth).length > 0 ||
+        getEventsForMonth(events, contentYear, contentMonth).length > 0 ||
+        getEventsForMonth(events, year, month).length > 0 ||
+        getFutureEvents(events, year, month, 1).length > 0 ||
+        getProposalsForMonth(proposals, contentYear, contentMonth).length > 0
+    );
```
Comment (mashehu) [inline sites/main-site/utils/newsletter.ts]:
```suggestion
        const checks = [
        () => getBlogPostsForMonth(blogPosts, contentYear, contentMonth),
        () => getBlogPostsForPreviousMonths(blogPosts, contentYear, contentMonth, 2),
        () => getAdvisoriesForMonth(advisories, contentYear, contentMonth),
        () => getAdvisoriesForPreviousMonths(advisories, contentYear, contentMonth, 2),
        () => getPipelineReleasesForMonth(pipelines, contentYear, contentMonth),
        () => getNewPipelines(pipelines, contentYear, contentMonth),
        () => getEventsForMonth(events, contentYear, contentMonth),
        () => getEventsForMonth(events, year, month),
        () => getFutureEvents(events, year, month, 1),
        () => getProposalsForMonth(proposals, contentYear, contentMonth),
    ];

    return checks.some((check) => check().length > 0);
```
is a bit easier to read for me.

**Review by mashehu — APPROVED** [review]

File: sites/main-site/utils/newsletter.ts (hunk 1 above)
Comment (ewels) [reply sites/main-site/utils/newsletter.ts]: Agreed 😅 — swapped the manual `month === 1 ? …` wraparound for the existing date-fns-based `adjacentMonths()` helper, so the year boundary is handled the same way as everywhere else in this file. Done in 54165a3.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: sites/main-site/utils/newsletter.ts (hunk 2 above)
Comment (ewels) [reply sites/main-site/utils/newsletter.ts]: Applied the `checks` array + `.some()` refactor — agree it reads better. Done in 54165a3.

---
_Generated by [Claude Code](https://claude.ai/code)_
